pkg/utils/hash.go

```diff
@@ -28,3 +34,8 @@ func DeepHashObject(hasher hash.Hash, objectToWrite interface{}) {
 	}
 	printer.Fprintf(hasher, "%#v", objectToWrite)
 }
+
+func HashNameNamespace(name, namespace string) string {
+	bytes := []byte(namespace + name)
+	return string(sha256.New().Sum(bytes)[:hashLength])
+}
```

Comment:
Thanks for picking this up, I brought it across incorrectly from the previous PR that I closed. I think that longer term we are going to need this functionality, because we are running into too many issues around conflicting resource names, and the lengths of resource names. 

Consulting [this](https://gobyexample.com/sha256-hashes) link, it looks like what we want and should generate valid kubernetes names. I'm finishing some testing elsewhere right now, but I'll try to implement this again and see if I can make the names valid.

Reply:
Hopefully this is fixed now, I've just pushed the change and will watch CI.